staging/src/k8s.io/apiserver/pkg/endpoints/installer.go

```diff
@@ -506,6 +511,33 @@ func (a *APIInstaller) registerResourceHandlers(path string, storage rest.Storag
 		}
 	}
 
+	var resourceInfo *storageversion.ResourceInfo
+	if utilfeature.DefaultFeatureGate.Enabled(features.StorageVersionAPI) &&
+		isStorageVersionProvider &&
+		storageVersionProvider.StorageVersion() != nil &&
+		// we don't need to migrate storageversions objects, because everytime
+		// the server is upgraded, these objects will be rewritten by the apiserver,
+		// they won't be stale
+		resource != "storageversions" {
```

Comment:
I didn't expect an override here.  With the right skip in the REST handler, we should be able to return this resourceinfo without an issue.

Reply:
Returning this resourceinfo won't affect the filter. But it will trigger the storage migrator to migrate storageversion objects, which is unnecessary.

Reply:
We need special handling for storageversions in one place or the other: 
1. skip collecting resourceinfo in API installer
2. skip blocking in the REST handler

1) saves the storage migrator from doing unnecessary migration (assuming kube-apiserver always writes the latest version of storageversion object after upgrade), but that may not hold if aggregated servers write storageversions. 

Let's go with 2). That makes the resource info collection more unified, and the REST handler control more centralized.